src/Common/Hooks/UseRegisterShortcut/UseRegisterShortcutProvider.tsx

```diff
@@ -14,19 +14,81 @@
  * limitations under the License.
  */
 
-import { useMemo, useState } from 'react'
+import { useCallback, useEffect, useMemo, useRef } from 'react'
+import { deepEquals } from '@rjsf/utils'
 import { context } from './UseRegisterShortcutContext'
-import { UseRegisterShortcutProviderType } from './types'
+import { UseRegisterShortcutProviderType, ShortcutType } from './types'
+import { preprocessKeys } from './utils'
+
+const ignoreTags = ['INPUT'].map((key) => key.toUpperCase())
 
 const UseRegisterShortcutProvider = ({ children }: UseRegisterShortcutProviderType) => {
-    const [registerShortcut, setRegisterShortcut] = useState(true)
+    const disableShortcuts = useRef<boolean>()
+    const shortcuts = useRef<Array<ShortcutType>>([])
+    const keysDown = useRef<string[]>([])
+
+    const registerShortcut = useCallback(({ keys, callback }: ShortcutType) => {
+        const processedKeys = preprocessKeys(keys)
```

Comment:
Suggestion: (Can discuss over it) if we have already registered a shortcut maybe can have a stack based approach for registeration?

Reply:
Discussed, please add a TODO

Reply:
added